src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
```

Comment:
```suggestion
	private static Map<String, Integer> CACHE = new ConcurrentHashMap<>();
```

Since this class may be used by different threads, I recommend changing the implementation to `ConcurrentHashMap`.

https://www.baeldung.com/java-concurrent-map